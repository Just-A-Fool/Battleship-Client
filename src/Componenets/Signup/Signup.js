import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Signup.css';

export default class Signup extends Component {

  render() {
    return (
      <form className>
        <div>
          <label htmlFor='signup-name-input'>Name</label>
          <input
            id='signup-name-input'
            name='name'
            required
          />
        </div>

        <div>
          <label htmlFor='signup-username-input'>Userame</label>
          <input
            id='signup-username-input'
            name='username'
            required
          />
        </div>

        <div>
          <label htmlFor='signup-password-input'>Password</label>
          <input
            id='signup-password-input'
            name='password'
            required
          />
        </div>

        <button type='submit'>Sign up</button>
        <button>
          <Link to='/guest'>
            Guest
          </Link>
        </button> <br />

        <Link to='/login'>Already have an account?</Link>
      </form>

      // <footer>
      // Copyright Â© since 2020
      // </footer>
    );
  };
  
};


