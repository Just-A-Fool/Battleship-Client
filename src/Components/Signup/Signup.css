.signupbtn {
  padding-top: 2rem;
}

.signup-errorMessage {
  color: red;
}

.signup-h1 {
  margin-top: 2rem;
}

