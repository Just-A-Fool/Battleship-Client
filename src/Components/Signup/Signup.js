import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Input, Required, Label } from '../Form/Form';
import AuthApiService from '../../Services/auth-api-service';
import Button from '../Button/Button';
import Banner from '../Banner/Banner';
import './Signup.css';

class Signup extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => { }
  };
  
  state = { error: null };
  firstInput = React.createRef();

  handleSubmit = ev => {
    ev.preventDefault()
    const { username, password, email } = ev.target

    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      email: email.value
    })
      .then(user => {
        email.value = ''
        username.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  };

  componentDidMount() {
    this.firstInput.current.focus()
  };

  render() {
    const { error } = this.state;
    return (
      <div>
        <Banner />
        
        <form className='signupform' onSubmit={this.handleSubmit}>
          <div role='alert'>
            { error && <p>{ error }</p>}
          </div>

          <div>
            <Label htmlFor='signup-email-input'>Enter your Email<Required /></Label>
            <Input
              ref={this.firstInput}
              id='signup-email-input'
              name='email'
              required
            />
          </div>

          <div>
            <Label htmlFor='signup-username-input'>Choose a username<Required /></Label>
            <Input
              id='signup-username-input'
              name='username'
              required
            />
          </div>

          <div>
            <Label htmlFor='signup-password-input'>Choose a password<Required /></Label>
            <Input
              id='signup-password-input'
              name='password'
              type='password'
              required
            />
          </div>

          <div className='signupbtn'>
            <Button type='submit'>Sign up</Button>
            {' '}

            <Button onClick={() => {
              this.props.history.push('/guest')
            }}> Guest
            </Button> <br />
          </div>

          <div className='btnLink'>
            <Link to='/login'>Already have an account?</Link>
          </div>
        
        </form>
      </div>
    );
  };
  
};

export default withRouter(Signup);


