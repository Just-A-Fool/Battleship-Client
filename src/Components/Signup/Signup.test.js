import React from 'react';
import ReactDOM from 'react-dom';
import Signup from './Signup';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

describe('Signup Component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter> 
        <Signup /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`Displays the signup page when rendered`, () => {
    const wrapper = mount(<BrowserRouter><Signup /></BrowserRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});


