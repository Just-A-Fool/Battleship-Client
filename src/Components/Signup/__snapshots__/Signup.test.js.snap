// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component Displays the signup page when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Signup)>
      <Route>
        <Signup
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          onRegistrationSuccess={[Function]}
        >
          <section
            className="signup"
          >
            <Banner>
              <div
                className="banner"
              >
                <h1>
                  BATTLESHIP
                </h1>
              </div>
            </Banner>
            <h1>
              Sign up
            </h1>
            <form
              className="signupform"
              onSubmit={[Function]}
            >
              <div>
                <Label
                  htmlFor="signup-username-input"
                >
                  <label
                    className="Label"
                    htmlFor="signup-username-input"
                  >
                    Choose a username
                    <Required>
                      <span
                        className="Required"
                      >
                        *
                      </span>
                    </Required>
                  </label>
                </Label>
                <ForwardRef
                  aria-required={true}
                  id="signup-username-input"
                  name="username"
                  required={true}
                >
                  <input
                    aria-required={true}
                    className="Input"
                    id="signup-username-input"
                    name="username"
                    required={true}
                    type="text"
                  />
                </ForwardRef>
              </div>
              <div>
                <Label
                  htmlFor="signup-password-input"
                >
                  <label
                    className="Label"
                    htmlFor="signup-password-input"
                  >
                    Choose a password
                    <Required>
                      <span
                        className="Required"
                      >
                        *
                      </span>
                    </Required>
                  </label>
                </Label>
                <ForwardRef
                  aria-required={true}
                  id="signup-password-input"
                  name="password"
                  required={true}
                  type="password"
                >
                  <input
                    aria-required={true}
                    className="Input"
                    id="signup-password-input"
                    name="password"
                    required={true}
                    type="password"
                  />
                </ForwardRef>
              </div>
              <div
                className="signupbtn"
              >
                <ForwardRef
                  type="submit"
                >
                  <button
                    className="Button"
                    type="submit"
                  >
                    Sign up
                  </button>
                </ForwardRef>
              </div>
              <div
                className="btnLink"
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    Already have an account?
                  </a>
                </Link>
              </div>
            </form>
          </section>
        </Signup>
      </Route>
    </withRouter(Signup)>
  </Router>
</BrowserRouter>
`;
